mittab/libs/data_import/import_teams.py

```diff
@@ -18,72 +18,147 @@
 #OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 #THE SOFTWARE.
 
-from tab.models import *
+from mittab.apps.tab.models import *
+
 import xlrd
-import csv
 from xlwt import Workbook
 
 def import_teams(fileToImport):
-    wb = xlrd.open_workbook(fileToImport)
-    sh = wb.sheet_by_index(0)
+    try:
+        sh = xlrd.open_workbook(filename=None, file_contents=fileToImport.read()).sheet_by_index(0)
+    except:
+        return ['ERROR: Please upload an .xlsx file. This filetype is not compatible']
     num_teams = 0
     found_end = False
+    team_errors = []
     while found_end == False:
         try:
             sh.cell(num_teams,0).value
             num_teams +=1
         except IndexError:
             found_end = True
-    for i in range(num_teams):
-        if i != 0:
-            team_school = sh.cell(i,1).value
+
+        #Verify sheet has required number of columns
+        try:
+            sh.cell(0, 8).value
+        except:
+            team_errors.append('ERROR: Insufficient Columns in Sheet. No Data Read')
+            return team_errors
+
+    for i in range(1, num_teams):
+
+        team_name = sh.cell(i,0).value
+        if team_name == '':
+            team_errors.append('Row ' + str(i) + ': Empty Team Name')
+            continue
+        try:
+            theName = Team.objects.get(name=team_name)
+            team_errors.append(team_name + ': Duplicate Team Name')
+            continue
+        except:
+            pass
+
+        team_school = sh.cell(i,1).value
```

Comment:
nit: spaces between arguments, e.g. i,1 => i, 1
(I know we're inconsistent elsewhere, not a big deal if you want to be like "meh")